### PR TITLE
feat: export mirror file with tls

### PR DESCRIPTION
**Pull request by fraidev**
Example: 
```
fluvio remote register user1 
fluvio remote export user1 --file user1.json --key ~/workspace/infinyon/fluvio/tls/certs/client-user1.key --cert ~/workspace/infinyon/fluvio/tls/certs/client-user1.crt
```

### CODE REVIEW

File: crates/fluvio-cli/src/client/remote/export.rs [hunk 1]
```diff
@@ -68,3 +80,79 @@ impl ExportOpt {
...
+                    domain: file_config.domain.clone(),
+                    ca_cert: std::fs::read_to_string(&file_config.ca_cert)?,
+                    client_cert: remote_cert,
+                    client_key: remote_key,
+                })),
+            }
+        }
+        _ => Ok(None),
+    }
+}
```
[the remark this answers is not in the document]
Comment (fraidev) [reply crates/fluvio-cli/src/client/remote/export.rs:148]: added

File: crates/fluvio-controlplane-metadata/src/mirror/spec.rs [hunk 2]
```diff
@@ -75,4 +75,25 @@ pub struct Home {
     pub id: String,
     pub remote_id: String,
     pub public_endpoint: String,
+    #[cfg_attr(feature = "use_serde", serde(skip_serializing_if = "Option::is_none"))]
```
[the remark this answers is not in the document]
Comment (fraidev) [reply crates/fluvio-controlplane-metadata/src/mirror/spec.rs:78]: done

File: k8-util/helm/fluvio-sys/templates/crd_mirror.yaml [hunk 3]
```diff
@@ -46,6 +46,17 @@ spec:
                           type: string
                         publicEndpoint:
                           type: string
+                        tls:
+                          type: object
```
[the remark this answers is not in the document]
Comment (fraidev) [reply k8-util/helm/fluvio-sys/templates/crd_mirror.yaml:50]: done

**Comment by fraidev** [comment]
> also should have CLI test as well

end to end with bats files?

**Review by sehz — APPROVED** [review]
LGTM

**Pull request merged**